src/header/js/header-menus.js

```diff
@@ -163,14 +163,23 @@ export default ({variant = 'desktop'}) => {
 
 								const hasStructured = hasCols && item.cols.filter((col) => (col.type === 'structured')).length > 0;
 								const structuredCols = hasCols ? item.cols.length : 0;
+								let multiCols = 0;
+
+								if (item.menus && item.menus.length > 10) {
+									(parseInt(item.menus.length % 3) === 0) ? multiCols = true : multiCols = false;
+								}
 
 								const $subcontent = $('div',
 									{
 										class: `${prefix}-submenu`,
 										id: `${prefix}-${variant}-submenu-${uuid}-${suuid}`,
 										'data-has-structured': hasStructured,
 										role: 'group', aria: {hidden: true, expanded: false},
-										data: {filled: (item.menus && item.menus.length > 10) ? item.menus.slice(0, 18).length : '', structuredCols: (structuredCols) ? structuredCols : ''}
+										data: {
+											filled: (item.menus && item.menus.length > 10) ? item.menus.slice(0, 30).length : '', 
+											structuredCols: (structuredCols) ? structuredCols : '', 
```

Comment:
You can remove `const structuredCols = hasCols ? item.cols.length : 0;` from line 165 and make this line the  following.

```js
structuredCols: hasCols ? item.cols.length : '',
```

Reply:
Awesome!! @gschnall

Reply:
@gschnall let me know.. did some refactoring as recommended.